### PR TITLE
Version catalog

### PR DESCRIPTION
**Pull request by ice1000**
Hopefully we can eliminate `deps`
/cc @imkiva need your help with project.version

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -11,16 +10,19 @@ plugins {
...
 var javaVersion: Int by rootProject.ext
-var deps: Properties by rootProject.ext
 
-// Remember to update .github/workflows/{nightly-build.yml, gradle-check.yml}
-javaVersion = 20
+projectVersion = libs.versions.project.get()
+javaVersion = libs.versions.java.get().toInt()
+
+// Workaround that `libs` is not available in `jacoco {}` block
+var jacocoVersion = libs.versions.jacoco.get()
```
Comment (imkiva) [inline build.gradle.kts:25]: Perfect workaround of `version.project` and `version.jacoco` 😉

**Review by imkiva — APPROVED** [review]

**Comment by imkiva** [comment]
Trying to make CI read the Java version from `libs.versions.toml`

**Comment by imkiva** [comment]
bors merge

**Comment by imkiva** [comment]
THE BIG GAME

image

**Comment by ice1000** [comment]
☺️ thank you!!

**Pull request merged**